portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,19 +43,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(convertToJson(messages));
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /** Get the message from the form and add it to the array */
+    String message = request.getParameter("text-input");
+    messages.add(message);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
   /**
    * Converts an array into a JSON string using manual String concatentation.
    */
   private String convertToJson(ArrayList messages) {
     String json = "{";
-    json += "\"string1\": ";
-    json += "\"" + messages.get(0) + "\"";
-    json += ", ";
-    json += "\"string2\": ";
-    json += "\"" + messages.get(1) + "\"";
-    json += ", ";
-    json += "\"string3\": ";
-    json += "\"" + messages.get(2) + "\"";
+    int i;
+    for (i = 0; i < messages.size() - 1; i++) {
```

Comment:
You can declare 'int i' within the for loop, so 'for (int i = 0; i < ...)'

Reply:
The reason I declared it outside of the for loop is because I use it again after the for loop to access the last element. I could also write 'messages.size() - 1' instead of 'i', and therefore, I would not need to use it outside the scope of the for loop. Should I make that change?